tests/scripts/run-pmi-diffs.py

```diff
@@ -0,0 +1,704 @@
+    else:
+        diff_root = os.path.abspath(diff_root)
+        validate_arg(diff_root, lambda item: os.path.isdir(diff_root))
+
+    if scratch_root is None:
+        scratch_root = os.path.join(diff_root, '_')
+    else:
+        scratch_root = os.path.abspath(scratch_root)
+
+    if ci_arch is not None:
+        validate_arg(ci_arch, lambda item: item in valid_ci_archs)
+
+    args = (arch, ci_arch, build_type, base_root, diff_root, scratch_root, skip_baseline_build, skip_diffs, target_branch, commit_hash)
+
+    log('Configuration:')
+    log(' arch: %s' % arch)
+    log(' ci_arch: %s' % ci_arch)
+    log(' build_type: %s' % build_type)
+    log(' base_root: %s' % base_root)
+    log(' diff_root: %s' % diff_root)
+    log(' scratch_root: %s' % scratch_root)
+    log(' skip_baseline_build: %s' % skip_baseline_build)
+    log(' skip_diffs: %s' % skip_diffs)
+    log(' target_branch: %s' % target_branch)
+    log(' commit_hash: %s' % commit_hash)
+
+    return args
+
+def nth_dirname(path, n):
+    """ Find the Nth parent directory of the given path
+    Args:
+        path (str): path name containing at least N components
+        n (int): num of basenames to remove
+    Returns:
+        outpath (str): path with the last n components removed
+    Notes:
+        If n is 0, path is returned unmodified
+    """
+
+    assert n >= 0
+
+    for i in range(0, n):
+        path = os.path.dirname(path)
+
+    return path
+
+def log(message):
+    """ Print logging information
+    Args:
+        message (str): message to be printed
+    """
+
+    print '[%s]: %s' % (sys.argv[0], message)
+
+def copy_files(source_dir, target_dir):
+    """ Copy any files in the source_dir to the target_dir.
+        The copy is not recursive.
+        The directories must already exist.
+    Args:
+        source_dir (str): source directory path
+        target_dir (str): target directory path
+    Returns:
+        Nothing
+    """
+
+    global testing
+    assert os.path.isdir(source_dir)
+    assert os.path.isdir(target_dir)
+
+    for source_filename in os.listdir(source_dir):
+        source_pathname = os.path.join(source_dir, source_filename)
+        if os.path.isfile(source_pathname):
+            target_pathname = os.path.join(target_dir, source_filename)
+            log('Copy: %s => %s' % (source_pathname, target_pathname))
+            if not testing:
+                shutil.copy2(source_pathname, target_pathname)
+
+##########################################################################
+# Do baseline build:
+# 1. determine appropriate commit,
+# 2. clone coreclr,
+# 3. do build
+##########################################################################
+
+def baseline_build():
+
+    if not testing:
+        if os.path.isdir(baseCoreClrPath):
+            log('Removing existing tree: %s' % baseCoreClrPath)
+            shutil.rmtree(baseCoreClrPath, onerror=del_rw)
+
+    # Find the baseline commit
+
```

Comment:
So this is currently diffing against a build at the head of the target branch?

Given how github merges up PRs that probably works well enough most of the time.